commit acb468a1805055e9f34dc1057eea186335bc2e13
Author: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date:   Fri Dec 9 15:53:05 2022 -0800

    cmake/Modules/Find{DSDP,MOSEK}.cmake: Do not hardcode blas/lapack lib names here

commit 8c8f7f82a16af70e044526f2911a94429d65b588
Author: Matthias Koeppe <mkoeppe@math.ucdavis.edu>
Date:   Fri Dec 9 19:09:51 2022 -0800

    CMakeLists.txt, src/CMakeLists.txt: Explicitly link through with blas


diff --git a/cmake/Modules/FindDSDP.cmake b/cmake/Modules/FindDSDP.cmake
index 8f43ae8..376fbdb 100644
--- a/cmake/Modules/FindDSDP.cmake
+++ b/cmake/Modules/FindDSDP.cmake
@@ -8,7 +8,7 @@ find_library(DSDP_LIBRARY
     HINTS ${DSDP_DIR} $ENV{DSDP_DIR}
     PATH_SUFFIXES lib)
 
-set(DSDP_LIBRARIES ${DSDP_LIBRARY} -llapack -lblas)
+set(DSDP_LIBRARIES ${DSDP_LIBRARY})
 
 include(FindPackageHandleStandardArgs)
 find_package_handle_standard_args(DSDP DEFAULT_MSG DSDP_INCLUDE_DIRS DSDP_LIBRARIES)
diff --git a/cmake/Modules/FindMOSEK.cmake b/cmake/Modules/FindMOSEK.cmake
index a8ee1d6..e3b7ab7 100644
--- a/cmake/Modules/FindMOSEK.cmake
+++ b/cmake/Modules/FindMOSEK.cmake
@@ -15,10 +15,10 @@ find_library(IOMP5_LIBRARY
     PATH_SUFFIXES bin)
 
 if(IOMPS_LIBRARY)
-  set(MOSEK_LIBRARIES ${MOSEK_LIBRARY} ${IOMP5_LIBRARY} -llapack -lblas -pthread)
+  set(MOSEK_LIBRARIES ${MOSEK_LIBRARY} ${IOMP5_LIBRARY} -pthread)
 else()
   # if libiomps is not available, we skip it
-  set(MOSEK_LIBRARIES ${MOSEK_LIBRARY} -llapack -lblas -pthread)
+  set(MOSEK_LIBRARIES ${MOSEK_LIBRARY} -pthread)
 endif()
 
 include(FindPackageHandleStandardArgs)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d4f14f5..fd2bdf9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -86,6 +86,12 @@ if(LAPACK_FOUND)
 else()
   message(FATAL_ERROR "Lapack not found")
 endif()
+find_package(BLAS REQUIRED)
+if(BLAS_FOUND)
+  message(STATUS "Found blas library: " ${BLAS_LIBRARIES})
+else()
+  message(FATAL_ERROR "Blas not found")
+endif()
 
 # search the selected symmetry computation program
 message(STATUS "Finding symmetry computation program \"${SYM}\"")
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 5e5d7d3..b3e2be9 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -125,11 +125,11 @@ setLibProperties(libscipsdp "scipsdp")
 #put binary in bin directory
 set (CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 if(SHARED)
-    target_link_libraries(libscipsdp PRIVATE ${ZIMPL_PIC_LIBRARIES} ${SDPS_PIC_LIBRARIES} ${SCIP_LIBRARIES} ${SYM_LIBRARIES} ${LAPACK_LIBRARIES} m)
+    target_link_libraries(libscipsdp PRIVATE ${ZIMPL_PIC_LIBRARIES} ${SDPS_PIC_LIBRARIES} ${SCIP_LIBRARIES} ${SYM_LIBRARIES} ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES} m)
     add_executable(scipsdp scipsdp/main.c ${scipsdpsources} ${objscipsdpsources} ${sdpisources} ${sym})
-    target_link_libraries(scipsdp ${ZLIB_LIBRARIES} ${Readline_LIBRARY} ${GMP_LIBRARIES} ${ZIMPL_LIBRARIES} ${SDPS_LIBRARIES} ${SCIP_LIBRARIES} ${SYM_LIBRARIES} ${LAPACK_LIBRARIES} m)
+    target_link_libraries(scipsdp ${ZLIB_LIBRARIES} ${Readline_LIBRARY} ${GMP_LIBRARIES} ${ZIMPL_LIBRARIES} ${SDPS_LIBRARIES} ${SCIP_LIBRARIES} ${SYM_LIBRARIES} ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES} m)
 else()
-    target_link_libraries(libscipsdp PRIVATE ${ZIMPL_LIBRARIES} ${SDPS_LIBRARIES} ${SYM_LIBRARIES} ${LAPACK_LIBRARIES} m)
+    target_link_libraries(libscipsdp PRIVATE ${ZIMPL_LIBRARIES} ${SDPS_LIBRARIES} ${SYM_LIBRARIES} ${LAPACK_LIBRARIES} ${BLAS_LIBRARIES} m)
     add_executable(scipsdp scipsdp/main.c ${scipsdpsources} ${objscipsdpsources} ${sdpisources} ${sym})
     target_link_libraries(scipsdp libscipsdp ${SCIP_LIBRARIES})
 endif()
